packages/backend/src/services/CsvService/CsvService.ts

```diff
@@ -200,7 +201,12 @@ export class CsvService {
             `writeRowsToFile with ${rows.length} rows and ${selectedFieldIds.length} columns`,
         );
 
-        const fileId = `csv-${nanoid()}.csv`;
+        const timestamp = moment().format('YYYY-MM-DD-HH-mm-ss-SSSS');
+        const sanitizedFileName = fileName
+            .toLowerCase()
+            .replace(/[^a-z0-9]/gi, '_') // Replace non-alphanumeric characters with underscores
+            .replace(/_{2,}/g, '_'); // Replace multiple underscores with a single one
+        const fileId = `csv-${sanitizedFileName}-${timestamp}.csv`;
```

Comment:
we could have this as part of a util in `CsvService` 😉 and some tests maybe? 

but non-blocking

Reply:
Good idea. Done